packages/grafana-data/src/types/pluginExtensions.ts

```diff
@@ -43,14 +53,23 @@ export type PluginExtensionConfig<Context extends object = object, ExtraProps ex
     ) => Partial<{ title: string; description: string } & ExtraProps> | undefined;
   };
 
-export type PluginExtensionLinkConfig<Context extends object = object> = PluginExtensionConfig<
+export type PluginExtensionLinkConfig<Context extends object = object> = PluginExtensionBaseConfig<
   Context,
   Pick<PluginExtensionLink, 'path'> & {
     type: PluginExtensionTypes.link;
     onClick?: (event: React.MouseEvent | undefined, helpers: PluginExtensionEventHelpers<Context>) => void;
   }
 >;
 
+export type PluginExtensionComponentConfig<Context extends object = object> = PluginExtensionBaseConfig<
+  Context,
+  Pick<PluginExtensionComponent, 'component'> & {
+    type: PluginExtensionTypes.component;
+  }
+>;
```

Comment:
I think it would be worth doing some repetition instead of creating the types out of other types with `Pick`. But also something we can do in a follow up PR.

Reply:
I think that's a good point, like this it's becoming quite cryptic 👍

Reply:
I tidied them up [in this commit](https://github.com/grafana/grafana/pull/68600/commits/e6a4beccd1c7295f81489bace38b6118843aadc9), ket me know what you think?

Reply:
Spot on! Great work on this.